### PR TITLE
useAsyncCallback

### PR DESCRIPTION
**Pull request by wardoost**
`useAsync` with a callback

Alternative names: useAsyncManual, useAsyncCall, useAsyncFn

### CODE REVIEW

File: docs/useAsyncCallback.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+## Usage
+
+```jsx
+import {useAsyncCallback} from 'react-use';
+
+const Demo = (url) => {
+  const [state, fetch] = useAsyncCallback(async () => {
+    const response = await fetch(url);
+    const result = response.text();
```
Comment (streamich) [inline docs/useAsyncCallback.md]: Should it be

```js
const result = await response.text();
```

**Comment by wardoost** [comment]
@streamich `useAsync` could definitely reuse this hook, will push that soon 👍 

Would it make sense to rename or alias `useAsync` to `useAsyncEffect`?

**Comment by streamich** [comment]
> Would it make sense to rename or alias `useAsync` to `useAsyncEffect`?

Maybe, or call this hook `useAsyncEffect`, up to you. But better *alias* so it is not a breaking change just because of this naming.

**Comment by wardoost** [comment]
I would not call this hooks `useAsyncEffect` as it's not an effect. Adding unnecessary aliases makes it harder for new users to know which hook they need. I think it's maybe best to rename this hook to `useAsyncFn` to align it with for example `useThrottleFn`.

**Pull request merged**